### PR TITLE
[CELEBORN-692] WorkerStatusTracker should handle WORKER_SHUTDOWN properly

### PR DESCRIPTION
**Pull request by waitinfuture**
### What changes were proposed in this pull request?
This PR put workers with WORKER_SHUTDOWN status into shuttingWorkers instead of blacklist.

### Why are the changes needed?
If WORKER_SHUTDOWN workers are put into blacklist, it will not trigger commit files, see ```CommitHandler::commitFiles```

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Manual test.

### CODE REVIEW

**Comment by waitinfuture** [comment]
cc @pan3793 @AngersZhuuuu @RexXiong

File: client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala [hunk 1]
```diff
@@ -123,16 +123,20 @@ class WorkerStatusTracker(
...
-                StatusCode.NO_AVAILABLE_WORKING_DIR |
-                StatusCode.RESERVE_SLOTS_FAILED |
-                StatusCode.UNKNOWN_WORKER =>
-              blacklist.put(worker, (statusCode, blacklist.get(worker)._2))
-            case _ => // Not cover
+          if (!blacklist.containsKey(worker)) {
+            blacklist.put(worker, (statusCode, registerTime))
+          } else {
+            statusCode match {
+              case StatusCode.WORKER_SHUTDOWN |
```
Comment (pan3793) [inline client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala:133]: possible?

Comment (waitinfuture) [reply client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala]: fixed, thanks

**Comment by pan3793** [comment]
The code can be simplified by pattern match to avoid nested if-else
```
      failedWorker.asScala.foreach {
        case (worker, (StatusCode.WORKER_SHUTDOWN, _)) =>
          shuttingWorkers.add(worker)
        case (worker, (statusCode, registerTime)) if !blacklist.containsKey(worker) =>
          blacklist.put(worker, (statusCode, registerTime))
        case (worker, (statusCode, _))
            if Array(
              StatusCode.NO_AVAILABLE_WORKING_DIR,
              StatusCode.RESERVE_SLOTS_FAILED,
              StatusCode.UNKNOWN_WORKER).contains(statusCode) =>
          blacklist.put(worker, (statusCode, blacklist.get(worker)._2))
        case _ =>
```

**Review by pan3793 — APPROVED** [review]
The logic LGTM. Based on my experience, the flatten `if` - `else if` - `else` combine with pattern match is readable than nested `if`-`else`. But this logic is not complex, either is fine to me.

**Comment by waitinfuture** [comment]
> 

Refined, thanks :)

**Pull request closed without merging**